pkg/resourceapply/batch.go

```diff
@@ -16,7 +17,26 @@ func ApplyJobWithControl(
 	required *batchv1.Job,
 	options ApplyOptions,
 ) (*batchv1.Job, bool, error) {
-	return ApplyGeneric[*batchv1.Job](ctx, control, recorder, required, options)
+	return ApplyGenericWithHandlers[*batchv1.Job](
+		ctx,
+		control,
+		recorder,
+		required,
+		options,
+		nil,
+		func(required *batchv1.Job, existing *batchv1.Job) string {
+			if !equality.Semantic.DeepEqual(existing.Spec.Template, required.Spec.Template) {
+				return "spec.template is immutable"
+			}
+			if !equality.Semantic.DeepEqual(existing.Spec.Completions, required.Spec.Completions) {
+				return "spec.completions is immutable"
+			}
+			if required.Spec.ManualSelector != nil && *required.Spec.ManualSelector && !equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector) {
```

Comment:
why the manual selector? I don't see that in
https://github.com/kubernetes/kubernetes/blob/fc69084bf1fd32e55256795a6db7a74d75679d8b/pkg/apis/batch/validation/validation.go#L377

Reply:
aligned with kube admission.